drivers/spi/spi_ll_stm32.c

```diff
@@ -679,6 +679,33 @@ static int transceive(const struct device *dev,
 
 #ifdef CONFIG_SPI_STM32_DMA
 
+#ifdef CONFIG_SOC_SERIES_STM32H7X
+
+static struct dma_mem_region stm32_mem_regions[] = {
+	DT_FOREACH_STATUS_OKAY(mmio_sram, DT_GET_MEM_REGION)
+};
+
+/*
+ * Checks that all buffers in `bufs` are in a valid area to be used
+ * with DMA.
+ *
+ */
+static bool all_bufs_in_valid_dma_ram(const struct spi_buf_set *bufs)
```

Comment:
probably should go in spi.h

spi_buf_set_in_nocache(const struct spi_buf_set *bufs)

Reply:
I see what you mean, will do.

Reply:
Bear in mind though that moving this to the `spi.h` header will involve to add an `#include <drivers/dma.h` to it as well, as the body of the function uses the `dma_mem_region` type. I have considered moving all this to either `devicetree.h` or some header in `include/zephyr/devicetree/*` but no structs/functions are declared there, so it doesn't seem consistent.